renku/core/management/datasets.py

```diff
@@ -1400,6 +1403,26 @@ def extract_dataset(filepath):
         return download_to.parent, [download_to]
 
 
+def _filename_from_headers(request):
+    """Extract filename fromcontent-disposition headers if available."""
+    content_disposition = request.headers.get("content-disposition", None)
+
+    if not content_disposition:
+        return None
+    entries = content_disposition.split(";")
+    name_entries = [e.strip() for e in entries if e.strip().lower().startswith("filename*=")]
```

Comment:
`name_entries = next((e.strip() for e in entries if e.strip().lower().startswith("filename*=")), None)` is a bit cleaner.